### PR TITLE
recreate policies branch

### PR DESCRIPTION
**Pull request by AlekEagle**
Supersedes #4

### CODE REVIEW

File: TOS.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Cumulonimbus Terms of Service
+
+Last Updated: September 15th, 2023
+
+This document describes the usage of Cumulonimbus (which from here on out may be referred to as `We`, `Us`, and/or `the Service`) allowed by you. By using the Service you are agreeing to these terms. THIS SERVICE IS PROVIDED AS-IS WITH ABSOLUTELY NO WARRANTY IN ANY WAY, SHAPE, OR FORM. WE ARE NOT RESPONSIBLE FOR ANY LOSS OF DATA. FAILURE TO ABIDE BY THESE TERMS MAY RESULT IN YOUR ACCOUNT BEING DISABLED (BANNED) OR TERMINATED. ANY ACTION TAKEN AGAINST YOUR ACCOUNT WILL BE ACCOMPANIED WITH A REASON, WHETHER IT BE BY EMAIL OR ON THE SERVICE ITSELF.
```
Comment (sudokoko) [inline TOS.md]:
```suggestion
This document describes the usage of Cumulonimbus (which from here on out may be referred to as `We`, `Us`, and/or `the Service`) allowed by you. By using the Service you are agreeing to these terms. THIS SERVICE IS PROVIDED AS-IS WITH ABSOLUTELY NO WARRANTY IN ANY WAY, SHAPE, OR FORM. WE ARE NOT RESPONSIBLE FOR ANY LOSS OF DATA. FAILURE TO ABIDE BY THESE TERMS MAY RESULT IN YOUR ACCOUNT BEING DISABLED ("BANNED") OR TERMINATED. ANY ACTION TAKEN AGAINST YOUR ACCOUNT WILL BE ACCOMPANIED WITH A REASON, WHETHER IT BE BY EMAIL OR ON THE SERVICE ITSELF.
```

File: PRIVACY.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+When uploading files to the Service, We store the file, when it was uploaded, and how big it is.
+
+When signing in, We store what device created the session, when it was created, and when it expires.
+
+When making a request to the Service, Your IP address will be logged by Cloudflare, and Our origin servers for abuse and DoS/DDoS protection. Our servers logs are deleted after 30 days. For more information about Cloudflare's policies, visit [Cloudflare's Privacy Policy](https://www.cloudflare.com/privacypolicy/).
+
+## How We Store Your Data
+
+Everything is stored in Our Database which is located on Our origin server, Our origin servers are located in the United States. All of Your files are also located on the origin server, uploads in their original form are not cached by Cloudflare. All previews that are generated by the preview server are cached by Cloudflare. Previews are deleted from Cloudflare's cache after 30 days. Passwords are hashed and salted on Our origin server Our origin server is only accessible on Our internal network.
```
Comment (sudokoko) [inline PRIVACY.md]:
```suggestion
Everything is stored in Our database which is located on Our origin server, Our origin servers are located in the United States. All of Your files are also located on the origin server, uploads in their original form are not cached by Cloudflare. All previews that are generated by the preview server are cached by Cloudflare. Previews are deleted from Cloudflare's cache after 30 days. Passwords are hashed and salted on Our origin server. Our origin server is only accessible on Our internal network.
```

File: PRIVACY.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+The personal information collected is used to provide security and ensure usability for the Service. By using the Service You agree to the collection and use of information in accordance with this Privacy Policy.
+
+We will alert you of these policies and procedures being updated both by changing the "Last Updated" date above and by prompting you on the Service. If We deem the change great enough you may receive an email concerning the change of the Privacy Policy. While you will be notified of any changes to these policies and procedures We still strongly recommend that you periodically review these policies and procedures.
+
+We respect You and Your privacy, and collect the smallest amounts of data We can.
+
+## What We Collect and Why
+
+When You create an account You provide a username, email address, and password. We use Your email address to contact You about any/all issues regarding Your account, and is also used for notifying You of Your completion of a data request.
```
Comment (sudokoko) [inline PRIVACY.md]:
```suggestion
When You create an account, You provide a username, email address, and password. We use Your email address to contact You about any/all issues regarding Your account, and is also used for notifying You of Your completion of a data request.
```

File: TOS.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+If you do not agree to these terms then usage must be discontinued immediately.
```
Comment (sudokoko) [inline TOS.md]:
```suggestion
If you do not agree to these terms, usage of the Service must be discontinued immediately.
```

File: TOS.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+We will alert you of these terms being updated both by changing the "Last Updated" date above and by prompting you on the Service. If We deem the change great enough you may receive an email concerning the change of the Terms of Service. While you will be notified of any changes to these terms We still strongly recommend that you periodically review these terms.
```
Comment (sudokoko) [inline TOS.md]:
```suggestion
We will alert you of these terms being updated both by changing the "Last Updated" date above and by prompting you on the Service. If We deem the change great enough, you may receive an email concerning the change of the Terms of Service. While you will be notified of any changes to these terms, We still strongly recommend that you periodically review these terms.
```

File: TOS.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+The source code of the Service is provided not to be used to create a hostile clone (duplicates of the Service to mislead, deceive, or act in any other malicious manner). Using the source code of the Service to create a hostile clone will result in defensive action.
```
Comment (sudokoko) [inline TOS.md]:
```suggestion
The source code of the Service is provided not to be used to create a "hostile clone" (duplicates of the Service to mislead, deceive, or act in any other malicious manner). Using the source code of the Service to create a "hostile clone" as described above will result in defensive action.
```

File: TOS.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+## Acceptable Use
+
+Any and all content uploaded by you is, and will always be your property. the Service is not liable for any user-generated content. The upload of illegal, harmful, and/or immoral content as defined within your jurisdiction, the state laws of Alaska and Missouri, and federal US laws is not allowed and will be removed with valid reason being sent to you via email. Continued violations may result in your account being disabled or terminated. Account termination will result in your account being deleted and all user generated content being deleted. Violations resulting in an account being disabled can be appealed, while account termination is permanent and cannot be reversed. Abuse of the API is not allowed and can result in your account being disabled or terminated.
```
Comment (sudokoko) [inline TOS.md]:
```suggestion
Any and all content uploaded by you is, and will always be Your property. the Service is not liable for any user-generated content. The upload of illegal, harmful, and/or immoral content as defined within Your jurisdiction, the state laws of Alaska and Missouri, and U.S. Federal Law is not allowed and will be removed with valid reason being sent to you via email. Continued violations may result in your account being disabled ("banned") or terminated. Account termination will result in Your account being deleted, as well as all user-generated content contained within Your account. Violations resulting in an account being disabled ("banned") can be appealed, while account termination is permanent and cannot be reversed. Abuse of the API is not allowed and can result in Your account being disabled ("banned") or terminated.
```

File: TOS.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+## Digital Millennium Copyright Act (DMCA) Claims
+
+If a user is to upload content that violates DMCA a request for a takedown can be filed at dmca@alekeagle.com with a subject of "Cumulonimbus DMCA Claim". Abuse of a DMCA Claim is not tolerated and will go ignored.
```
Comment (sudokoko) [inline TOS.md]:
```suggestion
If a user is to upload content that violates DMCA, a request for a take-down can be filed at dmca@alekeagle.com with a subject of "Cumulonimbus DMCA Claim". Abuse of a DMCA Claim is not tolerated and will go ignored.
```

File: TOS.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+## Contact
+
+If you have any questions please contact Us at contact@alekeagle.com
```
Comment (sudokoko) [inline TOS.md]:
```suggestion
If You have any questions please contact Us at contact@alekeagle.com
```

**Review by sudokoko — COMMENTED** [review]
grammatical nitpicks

**Review by sudokoko — APPROVED** [review]
this looks fine to me

File: PRIVACY.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+## General Data Protection Regulation (GDPR) Requests
+
+Outside of logs, all data We store related to Your account can be accessed via Your dashboard. If you'd like to access all logs related to Your account, you can reach out to <gdpr@alekeagle.com> using the email attributed to Your account, in which you will receive Your information in 2 business days.
+
+Anyone may make a request, irregardless if you live under a jurisdiction with the GDPR or CCPA regulations.
+
+## Contact
+
+If you have any questions please contact Us at <contact@alekeagle.com>.
```
Comment (sudokoko) [inline PRIVACY.md]: whoops missed this
```suggestion
If You have any questions please contact Us at <contact@alekeagle.com>.
```

**Comment by AlekEagle** [comment]
Honestly, even though logs have IP addresses, I have no way to identify whose requests belong to who when it comes to getting logs for GDPR requests.

**Comment by sudokoko** [comment]
> Honestly, even though logs have IP addresses, I have no way to identify whose requests belong to who when it comes to getting logs for GDPR requests.

One way that you could do this, inspired from how ASP.NET traces requests, is assign a user a unique ID that is logged alongside requests, that way you can easily `grep` for it.

My services log the individual request IDs that ASP.NET provides, they look something like this:
image

**Comment by AlekEagle** [comment]
> > Honestly, even though logs have IP addresses, I have no way to identify whose requests belong to who when it comes to getting logs for GDPR requests.
> 
> One way that you could do this, inspired from how ASP.NET traces requests, is assign a user a unique ID that is logged alongside requests, that way you can easily `grep` for it.
> 
> My services log the individual request IDs that ASP.NET provides, they look something like this:
> image
> 

That's a good idea, but there is one catch for this, the Cumulonimbus API only logs verbosely (e.g. `User a (1234567890) requested their own user object` ) under a development environment (`ENV=development`), the logs that I am particularly concerned about are NGINX's logs. NGINX logs are formatted like so:

```txt
IP(CF_X_FORWARDED_FOR) - [Local timestamp] HOST "METHOD REQUEST_URI HTTP/VER" RESPONSE_STATUS_CODE CONTENT_LENGTH "HTTP_REFERER" "USER_AGENT"
```

Fortunately/unfortunately, none of this information alone can be consistently related to an individual user without additional information and context. While I _could_ have NGINX log the token that a user provides for authorization so I could consistently relate API calls to an individual user, the idea of making NGINX log users' tokens, regardless of the security hardening to prevent unauthorized access to the system and these logs, is ludicrously insecure (in my opinion) and is not worth the security liabilities it can introduce.

To conclude, while it is possible to manually crawl through the NGINX access logs to obtain logs for a specific user, it's impractical and doesn't contain enough significant information for a user to justify being sent in a GDPR account data request (in my opinion).

TL;DR: at the current moment in time, there are no logs that contain pertinent information that necessitate GDPR data requests or data erasures.

**Comment by sudokoko** [comment]
> > > Honestly, even though logs have IP addresses, I have no way to identify whose requests belong to who when it comes to getting logs for GDPR requests.
> > 
> > 
> > One way that you could do this, inspired from how ASP.NET traces requests, is assign a user a unique ID that is logged alongside requests, that way you can easily `grep` for it.
> > My services log the individual request IDs that ASP.NET provides, they look something like this:
> > image
> 
> That's a good idea, but there is one catch for this, the Cumulonimbus API only logs verbosely (e.g. `User a (1234567890) requested their own user object` ) under a development environment (`ENV=development`), the logs that I am particularly concerned about are NGINX's logs. NGINX logs are formatted like so:
> 
> ```
> IP(CF_X_FORWARDED_FOR) - [Local timestamp] HOST "METHOD REQUEST_URI HTTP/VER" RESPONSE_STATUS_CODE CONTENT_LENGTH "HTTP_REFERER" "USER_AGENT"
> ```
> 

Makes sense. I'm not using NGINX for logging since I'm not using NGINX at all, as ASP.NET is it's own web server - so I wouldn't know lol. Me personally I use Serilog's ASP.NET request logging library - and my logs are only stored in the system journal which is rotated out regularly.

**Comment by AlekEagle** [comment]
I believe these policies have evolved to the point that they are ready for review and merge. If anyone has any suggestions to them please let me know, otherwise they will be merged into the master branch in a few hours, and pages will be created on the frontend for them.

**Review by sudokoko — APPROVED** [review]
Looks fine to me
